app/controllers/api/v2/preupgrade_reports_controller.rb

```diff
@@ -3,17 +3,32 @@
 module Api
   module V2
     class PreupgradeReportsController < ::Api::V2::BaseController
-      before_action :find_resource, only: %i[show]
-
       api :GET, '/preupgrade_reports/', N_('List Preupgrade reports')
       param_group :search_and_pagination, ::Api::V2::BaseController
       def index
         @preupgrade_reports = resource_scope_for_index
       end
 
-      api :GET, '/preupgrade_reports/:id', N_('Show Preupgrade report')
+      api :GET, '/job_invocations/:id/preupgrade_report', N_('Show Preupgrade report')
       param :id, :identifier, required: true
-      def show; end
+      def show
+        @preupgrade_report = PreupgradeReport.find_by(job_invocation_id: params[:id])
```

Comment:
A single job can have multiple preupgrade reports (N reports for N hosts in the job), right? In that case, this cannot use `find_by` which returns a single record.

We should probably have `/job_invocations/:job_id/preupgrade_reports` which shows all the preupgrade reports in the job and `/job_invocations/:job_id/preupgrade_reports/:host_id` which shows a single preupgrade report for a given host in the job

Reply:
Yeah that's true, I completely forgot that. How about splitting it into two actions:
```
/job_invocations/:job_id/preupgrade_reports (get all reports for the job_invocation)
/preupgrade_reports/:id (show report)
```

Reply:
Fine by me